File: src/test/java/learnyouakotlin/part4/SignupServer.java
package learnyouakotlin.part4;


import com.sun.net.httpserver.HttpServer;

import java.io.IOException;
import java.net.InetSocketAddress;
import java.util.concurrent.Executors;

/**
 * Run the signup handler with in-memory storage, for manual testing
 */
public class SignupServer {
    public static void main(String[] args) throws IOException {
        final var book = new InMemorySignupBook();
        for (int i = 1; i <= 10; i++) {
            book.save(new Available(SessionId.of(Integer.toString(i)), 20));
        }

        int port = 9876;
        final var server = HttpServer.create(new InetSocketAddress(port), 0);
        // So we don't have to worry that SignupSheet and SignupBook are not thread safe
        server.setExecutor(Executors.newSingleThreadExecutor());
        server.createContext("/", new SignupHttpHandler(new InMemoryTransactor<>(book)));
        server.start();

        System.out.println("Ready on port " + port);
    }
}

File: src/test/java/learnyouakotlin/part4/SignupServer.kt
@file:JvmName("SignupServer")

package learnyouakotlin.part4

import com.sun.net.httpserver.HttpServer
import org.glassfish.jersey.uri.UriTemplate
import java.net.InetSocketAddress
import java.util.concurrent.Executors
import java.util.function.Consumer


/**
 * Run the signup handler with in-memory storage, for manual testing
 */
fun main() {
    val book = InMemorySignupBook()
    
    for (i in 1..10) {
        val session = SignupSheet()
        session.sessionId = SessionId(i.toString())
        session.capacity = 20
        book.save(session)
    }
    
    val port = 9876
    val server = HttpServer.create(InetSocketAddress(port), 0)
    
    // So we don't have to worry that SignupSheet and SignupBook are not thread safe
    server.executor = Executors.newSingleThreadExecutor()
    server.createContext("/", SignupHttpHandler(InMemoryTransactor(book)))
    server.start()
    
    println("Ready:")
    SignupHttpHandler.routes.forEach(Consumer { template: UriTemplate ->
        println("- " + "http://localhost:" + port + template.template)
    })
}
